lib/tasks/geoip.rake

```diff
@@ -0,0 +1,33 @@
+namespace :geoip do
+  require 'open-uri'
+
+  desc 'Download the latest MaxMind geoip data file'
+  task download_data: :environment do
+    # download location as documented at:
+    #   https://dev.maxmind.com/geoip/geoip2/geolite2/
+    link = 'https://geolite.maxmind.com/download/geoip/database/GeoLite2-Country.tar.gz'
+    target_dir = "#{Rails.root}/files/geodata"
+    downloaded_location = File.join(Dir::tmpdir, 'geodata.tar.gz')
+
+    File.open(downloaded_location, "wb") do |saved_file|
+    # the following "open" is provided by open-uri
+      open(link, "rb") do |read_file|
+        saved_file.write(read_file.read)
+      end
+    end
+
+    `tar -xzf #{downloaded_location} -C #{Dir::tmpdir}`
+
+    extracted_folder = Dir["#{Dir::tmpdir}/GeoLite2-Country_*"].last
+    `mv #{extracted_folder}/GeoLite2-Country.mmdb #{target_dir}/GeoLite2-Country.mmdb`
```

Comment:
Could use [`FileUtils.mv`](https://ruby-doc.org/stdlib-2.5.3/libdoc/fileutils/rdoc/FileUtils.html#method-c-mv) to avoid shelling out and reduce line length